### PR TITLE
[PAPER] JOSS for v0.1/0.2 ICLabel port

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #59 

Draft PR to make sure everything is in order before we submit to JOSS via their Github.

Title: MNE-ICALabel: Automatically annotating ICA components with ICLabel in Python
Git branch containing paper: https://github.com/mne-tools/mne-icalabel/pull/61
Repo: https://github.com/mne-tools/mne-icalabel
Suggested Editor: Olivia Guest
Message to Editors: This is a submission of a Python software module that replicates the function of ICLabel for automatic ICA component labeling in EEG data, originally in EEGLab. It is made into a simple API that is compatible with the popular MNE-Python framework.
https://joss.theoj.org/papers/new

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-connectivity/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - affiliation: 3
+    name: Mathieu Scheltienne
+    orcid: tbd
+affiliations:
+- index: 1
+  name: Department of Computer Science, Columbia University, New York, United States
+- index: 2
+  name: Department of Biomedical Engineering, Johns Hopkins University, Baltimore, United States
+- index: 3
+  name: tbd
```
Comment (mscheltienne) [inline paper/paper.md:30]:
```suggestion
  name: Human Neuroscience Platform, Fondation Campus Biotech Geneva, Geneva, Switzerland
- index 4
  name: Center for Neuroprosthetics, Institute of Bioengineering, Ecole Polytechnique Fédérale de Lausanne (EPFL), Geneva, Switzerland
```

I'd like to add both (3) and (4) to my affiliations as I've spent time from both jobs on this project.
I have to double-check the EPFL (4) affiliation, I've got a doubt if I should point to the Center for Neuroprosthetics or CIBM.

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    orcid: 0000-0001-8421-365X
+  - affiliation: 2
+    name: Jacob Feitelberg
+    orcid: tbd
+  - affiliation: 3
+    name: Anand Saini
+    orcid: tbd
+  - affiliation: 3
+    name: Mathieu Scheltienne
+    orcid: tbd
```
Comment (mscheltienne) [inline paper/paper.md:23]:
```suggestion
    orcid: 0000-0001-8316-7436 
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  name: Department of Biomedical Engineering, Johns Hopkins University, Baltimore, United States
+- index: 3
+  name: tbd
+date: 18 May 2022
+bibliography: paper.bib
+---
+
+# Summary
+
+Scalp electroencephalography (EEG) and magnetoencephalography (MEG) analysis is typically very noisy and contains various non-neural signals, such as heart beat artifacts. Independent component analysis (ICA) is a common procedure to remove these artifacts [@Bell1995]. However, removing artifacts requires manual annotation of ICA components, which is subject to human error and very laborious when operating on large datasets. This work adds the popular ICLabel model [@iclabel2019] to the MNE-Python [@Agramfort2013] software toolkit. This enables the automatic labeling and annotation of ICA components improving the preprocessing and analysis pipeline of electrophysiological data.
```
Comment (mscheltienne) [inline paper/paper.md:37]:
```suggestion
Scalp electroencephalography (EEG) and magnetoencephalography (MEG) analysis is typically very noisy and contains various non-neural signals, such as heart beat artifacts. Independent component analysis (ICA) is a common procedure to remove these artifacts [@Bell1995]. However, removing artifacts requires manual annotation of ICA components, which is subject to human error and very laborious when operating on large datasets. This work adds the popular ICLabel model [@iclabel2019] to the MNE-Python [@Agramfort2013] software toolkit. This enables the automatic labeling of ICA components, improving the preprocessing and analysis pipeline of electrophysiological data.
```

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+# Statement of need
+
+Typically EEG and MEG data has many artifacts due to a variety of reasons, such as heart beats, eye blinks, line noise, channel noise, or muscle artifacts. A common signal processing technique for assisting in separating true brain signal from these types of signals is ICA [@Bell1995]. ICA performs blind-source separation decomposing the observed noisy electrophysiological signals into "sources", which are maximally independent. For example, this then allows researchers to decompose EEG signals into brain signals and a heart beat signal. Then by manually annotating a component time-series as a heart beat, one can remove the heart beat source from the signal and obtain their signals without the heart beat. This process typically is manually carried out and requires a human to label each component that comes out of ICA. This is subject to human error and is difficult to scale up when dealing with high-dimensional EEG, or MEG recordings.
```
Comment (mscheltienne) [inline paper/paper.md:43]:
```suggestion
Typically EEG and MEG data have many artifacts due to a variety of signal sources, such as heartbeats, eye blinks, line noise, channel noise, or muscle artifacts. A common signal processing technique for assisting in separating true brain signals from noise sources is ICA [@Bell1995]. ICA performs blind-source separation decomposing the observed noisy electrophysiological signals into maximally independent sources. For example, it allows researchers to decompose EEG signals into brain signals and a heartbeat signal. Then by manually annotating a component time-series as a heartbeat, one can remove the heartbeat source from the signal and obtain their signals without the heartbeat. This process is typically manually carried out and requires a human to label each component that comes out from an ICA decomposition. This process is subject to human error and is difficult to scale up when dealing with high-dimensional EEG or MEG recordings.
```

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+ICLabel was a proposed statistical model that uses neural networks and a crowdsourced training dataset to automatically label ICA components [@iclabel2019]. However, this package and functionality was previously only available in MATLAB, limiting its usage among Python neuroscience researchers. Moreover, the package relied on an outdated version of Tensorflow, which makes the model difficult to build on. In this work, we will build on top of the ICLabel model to improve its robustness when operating on different types of recording hardware and channel count. Moreover, we plan on building new types of models that improve its overall accuracy and performance with respect to auto-labeling neural and non-neural signals. The availability of a simple API for ICLabel will facilitate a strong benchmark to build future models and bring in new developers and contributors.
```
Comment (mscheltienne) [inline paper/paper.md:45]:
```suggestion
ICLabel was a proposed statistical model that uses neural networks and a crowdsourced training dataset to automatically label ICA components [@iclabel2019]. However, this package was previously only available in MATLAB, limiting its usage among Python neuroscience researchers. Moreover, the package relied on an outdated version of Tensorflow, which makes the model difficult to build on. In this work, we will build on top of the ICLabel model to improve its robustness when operating on different types of recording hardware, sensor type, and sensor count. Moreover, we plan on building new types of models that improve their overall accuracy and performance with respect to auto-labeling neural and non-neural signals. The availability of a simple API for ICLabel will facilitate a strong benchmark to build future models and bring in new developers and contributors.
```

Do you want to add the Tensorflow paper? I'm not sure it's necessary.

```
@inproceedings{45381,
title	= {TensorFlow: A system for large-scale machine learning},
author	= {Martin Abadi and Paul Barham and Jianmin Chen and Zhifeng Chen and Andy Davis and Jeffrey Dean and Matthieu Devin and Sanjay Ghemawat and Geoffrey Irving and Michael Isard and Manjunath Kudlur and Josh Levenberg and Rajat Monga and Sherry Moore and Derek G. Murray and Benoit Steiner and Paul Tucker and Vijay Vasudevan and Pete Warden and Martin Wicke and Yuan Yu and Xiaoqiang Zheng},
year	= {2016},
URL	= {https://www.usenix.org/system/files/conference/osdi16/osdi16-abadi.pdf},
booktitle	= {12th USENIX Symposium on Operating Systems Design and Implementation (OSDI 16)},
pages	= {265--283}
}
```

File: paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+MNE-Python is a general-purpose electrophysiology analysis package in Python that has a large core group of developers. Integration into MNE makes it likely that MNE-ICALabel will be maintained and continue to be improved. MNE-Python also has stability due to the funding it receives directly for development from institutions such as National Institutes of Health, the Chan Zuckerberg open-source initiative, the European Research Council and Agence Nationale de la Recherche in France.
+
+The developer team is excited to improve the state of the art in data handling and looking forward to welcoming new contributors and users.
+
+# Acknowledgements
+
+``MNE-ICALabel`` development is partly supported by
+the National Science Foundation under Grant # 2030859 to the Computing Research Association for the CIFellows Project.
```
Comment (mscheltienne) [inline paper/paper.md:54]:
```suggestion
``MNE-ICALabel`` development is partly supported by
the National Science Foundation under Grant # 2030859 to the Computing Research Association for the CIFellows Project and by the Human Neuroscience Platform, Fondation Campus Biotech
Geneva, Geneva, Switzerland.
```

Do you mind adding an acknowledgment to the FCBG for opening a 6-month internship position on this project?

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+We acknowledge the work of [@iclabel2019], which was originally produced in MATLAB in the popular EEGLab package.
+
+# References
```
Comment (mscheltienne) [inline paper/paper.md:58]: It feels to me like something should be written down here? 😅

File: paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  - affiliation: 1
+    name: Adam Li
+    orcid: 0000-0001-8421-365X
+  - affiliation: 2
+    name: Jacob Feitelberg
+    orcid: tbd
+  - affiliation: 3
+    name: Anand Saini
+    orcid: tbd
+  - affiliation: 3
```
Comment (mscheltienne) [inline paper/paper.md:21]:
```suggestion
  - affiliation: 3, 4
```

**Review by mscheltienne — CHANGES_REQUESTED** [review]
Looks good! Have a look at the proposed changes and let me know!

File: paper/paper.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+authors:
+  - affiliation: 1
+    name: Adam Li
+    orcid: 0000-0001-8421-365X
+  - affiliation: 2
+    name: Jacob Feitelberg
+    orcid: tbd
+  - affiliation: 3
+    name: Anand Saini
+    orcid: tbd
```
Comment (adam2392) [inline paper/paper.md:20]: @anandsaini024 what is your ORCId?

File: paper/paper.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  - iEEG
+  - ICA
+  - EEGLab
+authors:
+  - affiliation: 1
+    name: Adam Li
+    orcid: 0000-0001-8421-365X
+  - affiliation: 2
+    name: Jacob Feitelberg
+    orcid: tbd
```
Comment (adam2392) [inline paper/paper.md:17]: @jacobf18 what is your ORCID?

File: paper/paper.md (hunk 7 above)
Comment (adam2392) [reply paper/paper.md:58]: Nope actually! I think JOSS does some text automation in the backend.

File: paper/paper.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  - affiliation: 3
+    name: Anand Saini
```
Comment (adam2392) [inline paper/paper.md:19]:
```suggestion
    name: Anand Prakash Saini
```

File: paper/paper.md (hunk 9 above)
Comment (adam2392) [inline paper/paper.md:20]:
```suggestion
    orcid: 0000-0003-3595-5969
```

File: paper/paper.md (hunk 10 above)
Comment (adam2392) [inline paper/paper.md:17]:
```suggestion
    orcid: 0000-0002-4551-0245
```

**Review by mscheltienne — APPROVED** [review]
Looks good, anything else to add?

**Comment by adam2392** [comment]
I think this is done on my end.

Lmk @hoechenberger, @anandsaini024 @jacobf18 if you guys have any comments to add before we put this up to JOSS?

File: paper/paper.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+- index: 2
+  name: Department of Biomedical Engineering, Johns Hopkins University, Baltimore, United States
+- index: 3
+  name: Human Neuroscience Platform, Fondation Campus Biotech Geneva, Geneva, Switzerland
+- index: 4
+  name: Institute of Bioengineering, Center for Neuroprosthetics, EPFL, Geneva, Switzerland
+- index: 5
+  name: Department of Radiology and Medical Informatics, University of Geneva (UNIGE),
+Geneva, Switzerland
+- index: 10
```
Comment (mscheltienne) [inline paper/paper.md:39]:
```suggestion
- index: 6
```
Or am I missing something?

**Comment by adam2392** [comment]
Kay submitted to get a review!

**Pull request merged**